### ISSUE TITLE
Support for floating point textures

### ISSUE THREAD

**Issue by anlumo** [flutter/flutter#141289]
### Use case

This is an extension to #136025.

Fragment shaders sometimes need lookup tables, for example for color transformations. These tables usually use floating point, since that's the natural way math works on GPUs. GLSL handles this transparently, and APIs like OpenGL have support for this via floating point textures (`GL_RGBA32F`, `GL_R32F`, etc).

### Proposal

If Flutter adds support for sync texture construction, this should also allow supplying a floating point texture to enable these kinds of operations in fragment shaders.

**Comment by jonahwilliams** [comment]
As far as I know SkSL does not support this so we cannot add support (yet).

**Comment by timcreatedit** [comment]
Hi @jonahwilliams, I'm not super familiar with SkSL, but in the docs it seems like it supports sampling a 16bit float from a texture: https://skia.org/docs/user/sksl/#evaluating-sampling-other-skshaders

This feature would be really really helpful for my take on Liquid glass, as I would love to encode shape normals in a texture for reuse. Right now it is difficult because of the precision loss.

https://pub.dev/packages/liquid_glass_renderer

**Comment by sethladd** [comment]
We might want to bump the priority up to P2 because this is useful for folks building a downstream liquid glass, as well as just advanced effects in general.

After talking with the engine team, I learned that a way to re-triage issues is to remove the triaged label. So I'll remove that to allow our process to re-triage and re-evaluate.

**Comment by jtmcdole** [comment]
It looks like Skia added kRGBA_F16_SkColorType and kRGBA_F32_SkColorType in ~Jan 2025, so this could be possible. Jonah's original comment was correct at the time.

https://api.skia.org/SkColorType_8h.html#a3ea8ab47612dbc4801c18445176c6481a441d64d0389ec53c0032d2c180df6449

**Comment by jtmcdole** [comment]
(I'll let the team re-triage; my P2 was just "This seems more important now, but not top of the list")

**Comment by jonahwilliams** [comment]
I think there is some confusion in this issue. The texture format is the upload/transfer format. But the sampled data is always floating point, regardless of whether the upload format was floating point or fixed point.

**Comment by timcreatedit** [comment]
Yes, for me this is about the precision I can get from each channel in the sampler

**Comment by jonahwilliams** [comment]
In the meantime I'd recommend uniform array for lookup tables

**Comment by jonahwilliams** [comment]
We'd also need to support a different sampling function but I believe texelFetch and friends aren't universally supported

**Comment by sethladd** [comment]
> In the meantime I'd recommend uniform array for lookup tables

@timcreatedit , let us know what you think about the above recommendation

**Comment by timcreatedit** [comment]
> > In the meantime I'd recommend uniform array for lookup tables
> 
> [@timcreatedit](https://github.com/timcreatedit) , let us know what you think about the above recommendation

It's great for fixed size LUTs, but it doesn't cover my use-case of exporting a high-precision texture from one shader and passing it into another. 

Afaik, Flutter shaders also can't export high precision textures, so would that then be a separate issue?

**Comment by gaaclarke** [comment]
> Afaik, Flutter shaders also can't export high precision textures, so would that then be a separate issue?

@timcreatedit it sounds like you are suggesting compute shaders?  Could you make another issue for that, please.  It helps if you can give a concrete example of what it would allow you to express differently.  Bonus points if you tie it to liquid glass.  Thanks!
